src/hooks/use-interval.ts

```diff
@@ -0,0 +1,27 @@
+import { useEffect, useRef } from "react";
+
+/**
+ * Calls function using the interval determined as a param
+ * which can be dynamically changed.
+ * @param callback function to be called in certain intervals
+ * @param interval as a number in milliseconds
+ */
+function useInterval(callback: () => void, interval: number) {
+	const savedCallback = useRef<() => void>();
+
+	useEffect(() => {
+		savedCallback.current = callback;
+	}, [callback]);
+
+	useEffect(() => {
+		function tick() {
+			savedCallback.current();
+		}
+		if (interval !== null) {
```

Comment:
Types `number` and `null` have no strict overlap — is this check necessary?

Reply:
Yeah guess not. Check is now removed.